### PR TITLE
feature: watch cancel

### LINKED ISSUE #77: feature: support watch cancel
Since in etcd v3 watch would watch all changes before the timeout, compared to only watch the first change in etcd v2. Thus v3 also supports manually cancel the watch before the watch expires.

### PR DESCRIPTION
**Pull request by Yiyiyimu**
To support cancel watch when needed

fix #77

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
According to etcd issue [#7742](https://github.com/etcd-io/etcd/issues/7742#issuecomment-294184158), implement cancel watch by close http connection

**Comment by Yiyiyimu** [comment]
It seems the original implementation: `cancel_request` have not got implemented in etcd side, that it would return the same error as transmitting `create1_request` instead of the correct one `create_request`

**Comment by membphis** [comment]
@nic-chen pls take a look at this PR

**Comment by nic-chen** [comment]
need to resolve conflicts first. @Yiyiyimu

**Comment by Yiyiyimu** [comment]
@nic-chen Thank you! Conflict fixed.

File: lib/resty/etcd/v3.lua [hunk 1]
```diff
@@ -630,7 +646,9 @@ local function watch(self, key, attr)
     if not callback_fun then
         return nil, err
     end
-
+    if opts.need_cancel == true then
+        return callback_fun.func, callback_fun.http_cli
```
[the remark this answers is not in the document]
Comment (Yiyiyimu) [reply lib/resty/etcd/v3.lua:647]: could I put them into one table, and return like `res, err`?

File: lib/resty/etcd/v3.lua [hunk 2]
```diff
@@ -541,6 +542,15 @@ local function request_chunk(self, method, host, port, path, opts, timeout)
...
     end
+
+    if opts.need_cancel == true then
+        return {
+            func = read_watch,
+            http_cli = http_cli
+        }
+    else
+        return read_watch
+    end
```
Comment (Yiyiyimu) [inline lib/resty/etcd/v3.lua:553]: @nic-chen is there any needs to also change the return structure of this place

Comment (nic-chen) [reply lib/resty/etcd/v3.lua]: yes，good

**Pull request merged**